tools/prepare_mapping_v2.py

```
@@ -0,0 +1,199 @@
+    """Load a YAML file and ensure required structure exists."""
+    
+    if not os.path.exists(path):
+        raise FileNotFoundError(f"{label} file not found: \"{path}\"")
+    if not os.path.isfile(path):
+        raise IsADirectoryError(f"{label} path is not a file: \"{path}\"")
+
+    with open(path, "r", encoding="utf-8") as f:
+        try:
+            data = yaml.safe_load(f)
+        except yaml.YAMLError as e:
+            raise ValueError(f"{label} file is not valid YAML\n\t   {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)